### ISSUE TITLE
determine how to cache local copies of IIIF manifests & set it up

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/geniza#323]
Hopefully we can use djiffy for this; it may need some updating since it's been primarily used with book content. (Possibly also will require updating for python / django version compatibility)

Need to check with IIIF community if there are any best practices for caching IIIF manifests

**Comment by rlskoeser** [comment]
I asked about this on the IIIF slack; I got a lot of helpful suggestions, although more related to customizing manifests; nothing specific to caching.  

Documenting some of the suggestions from the slack conversation:
- In the cases where we are combining canvases from multiple manifests, we copy the canvas as-is from the original manifest and then can add a `partOf` relationship to the original manifest so we have the source documented. Some discussion about label; could be label from original manifest; perhaps prefixed with "original source" or similar. No viewers currently support displaying this, so it's really more internal documentation. Example:
```json
{
    "type": "Canvas",
    "partOf": [
        {
           "id": "https://example.com/a-manifest",
           "type": "Manifest",
           "label": { "en": ["original source: blah blah"] } 
        }
    ]
}
```
- canvas ids should be stable, so we should be able to write something that would check and update our cached canvases against any upstream changes
- We should include original provider information and logos; provider is newer (v3) and not yet supported by any viewers; example https://iiif.io/api/cookbook/recipe/0234-provider/
- We need to make sure we preserve any `requiredStatement` in the original manifest; that is per-manifest, not per-canvas; in the rare cases where we combine manifests from different institutions we'll have to figure out how to combine them.

**Comment by rlskoeser** [comment]
merged into develop in #405
